sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -801,4 +841,134 @@ inline pi_result piDevicePartition(
                                   phSubDevices, NumSubDevices));
   return PI_SUCCESS;
 }
+
+// Device
+///////////////////////////////////////////////////////////////////////////////
+
+///////////////////////////////////////////////////////////////////////////////
+// Context
+inline pi_result piContextCreate(const pi_context_properties *properties,
+                                 pi_uint32 num_devices,
+                                 const pi_device *devices,
+                                 void (*pfn_notify)(const char *errinfo,
+                                                    const void *private_info,
+                                                    size_t cb, void *user_data),
+                                 void *user_data, pi_context *retcontext) {
+  // TODO: Implement callback (if needed)
+  auto hDevices = reinterpret_cast<const ur_device_handle_t *>(devices);
+  auto phContext = reinterpret_cast<ur_context_handle_t *>(retcontext);
+
+  HANDLE_ERRORS(urContextCreate(num_devices, hDevices, nullptr, phContext));
+
+  return PI_SUCCESS;
+}
+
+inline pi_result piContextGetInfo(pi_context context,
+                                  pi_context_info param_name,
+                                  size_t param_value_size, void *param_value,
+                                  size_t *param_value_size_ret) {
+  static std::unordered_map<pi_context_info, ur_context_info_t> InfoMapping = {

```

Comment:
let's not use the map, as we know it introduces overhead. You may use the implementation from the UR L0 adapter branch, https://github.com/jandres742/llvm/blob/b9a7fbb4da41c8c09645011a9f6ff926fba77725/sycl/plugins/unified_runtime/pi2ur.hpp#L1055